director/director.go

```diff
@@ -148,11 +159,61 @@ func queryOrigins(ctx *gin.Context) {
 	ctx.JSON(http.StatusOK, res)
 }
 
+func handleFilterServer(ctx *gin.Context) {
+	sn := strings.TrimPrefix(ctx.Param("name"), "/")
+	if sn == "" {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": "name is a required path parameter"})
+		return
+	}
+	filtered, ft := checkFilter(sn)
+	if filtered {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": "Can't filter a server that already has been fitlered with type " + ft})
+		return
+	}
+	filteredServersMutex.Lock()
+	defer filteredServersMutex.Unlock()
+
+	// If we previously temporarily allowed a server, we switch to permFiltered (reset)
+	if ft == tempAllowed {
+		filteredServers[sn] = permFiltered
+	} else {
+		filteredServers[sn] = tempFiltered
+	}
+	ctx.JSON(http.StatusOK, gin.H{"message": "success"})
+}
+
+func handleAllowServer(ctx *gin.Context) {
```

Comment:
I'd prefer `handleAllowServer` --> `handlePermittedServer`, but again that's preference. I respect whatever judgement call you make!

Also, can you add a comment header with this functions intended behavior?

Reply:
I'd like to stick with Allow if that's not a big issue to make it consistent with the URL name, unless we'd like to change that too